pyzeebe/channel/oauth_channel.py

```diff
@@ -116,34 +152,38 @@ def create_camunda_cloud_channel(
         grpc.aio.Channel: The gRPC channel for connecting to Camunda Cloud.
     """
 
-    grpc_address = f"{cluster_id}.{region}.zeebe.camunda.io:443"
+    cluster_id = get_camunda_cloud_cluster_id(cluster_id)
+    region = get_camunda_cloud_cluster_region(region)
 
     oauth2_client_credentials = Oauth2ClientCredentialsMetadataPlugin(
-        client_id=client_id,
-        client_secret=client_secret,
-        authorization_server=authorization_server,
-        scope=scope,
-        audience=audience,
+        client_id=get_camunda_client_id(client_id),
+        client_secret=get_camunda_client_secret(client_secret),
+        authorization_server=get_camunda_oauth_url(authorization_server)
+        or "https://login.cloud.camunda.io/oauth/token",
+        scope=get_camunda_credentials_scopes(scope) or "Zeebe",
+        audience=get_camunda_token_audience(audience) or "zeebe.camunda.io",
         leeway=leeway,
         expire_in=expire_in,
     )
 
-    # NOTE: Overwrite the _oauth.fetch_token method to include client_id, client_secret in the request body
-    func = partial(
-        oauth2_client_credentials._oauth.fetch_token,
+    # NOTE: Overwrite the _oauth.fetch_token method to include client_id in the request body
+    oauth2_client_credentials._func_retrieve_token = partial(
+        oauth2_client_credentials._func_retrieve_token,
         include_client_id=True,
-        token_url=authorization_server,
-        client_secret=client_secret,
-        audience=audience,
     )
```

Comment:
This is not working. Please don't change the partial. Revert.

Reply:
Do you know why? I am pretty new to python and as far as I have seen, those parameters were already applied to `oauth2_client_credentials._func_retrieve_token`.

Reply:
Yes because it expects a partial in the definition and you are specifiying a partial with the partial expecting function. (circular)

Instead the oauth2session.fetch_token should be parametrized with the partial to prepare it to be executed before every (not really) rpc.

For more specifics look at the class `Oauth2ClientCredentialsMetadataPlugin`

Reply:
@PSanetra You are correct with the testing, I could add some more specific unit tests for the partial, that is true. Though that won't solve the main problem.

The problem is the behaviour for specific parameters can really only be testet with integration tests, which for Camunda Cloud (SaaS) i do locally, but it isn't possible for this repo, because there is no pyzeebe Camunda Cloud Cluster...

Reply:
I have reverted this change